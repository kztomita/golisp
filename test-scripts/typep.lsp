(print (typep 1 'integer))
;(print (typep 1 '(integer 0 100)))
(print (typep 1 'rational))
(print (typep 1 'real))
(print (typep 1 'number))
(print (typep 1 'float))
(print (typep 1 t))
(print (typep 1 nil))
(print "====")
(print (typep 1.0 'float))
(print (typep 1.0 'real))
(print (typep 1.0 'number))
(print (typep 1.0 'integer))
(print (typep 1.0 t))
(print (typep 1.0 nil))
(print "====")
(print (typep "foo" 'string))
(print (typep "foo" 'vector))
(print (typep "foo" 'array))
(print (typep "foo" 'sequence))
(print (typep "foo" 'symbol))
(print "====")
(print (typep 'foo 'symbol))
(print (typep 'foo 'string))
(print "====")
(print (typep :foo 'keyword))
(print (typep :foo 'symbol))
(print (typep :foo 'string))
(print "====")
(print (typep '(foo) 'cons))
(print (typep '(foo) 'list))
(print (typep '(foo) 'sequence))
(print "====")
(print (typep nil 'null))
(print (typep nil 'symbol))
(print (typep nil 'list))
(print (typep nil 'sequence))
(print (typep nil 'cons))
(print "====")
(print (typep t 'symbol))
(print "====")
(print (typep (lambda ()) 'function))

