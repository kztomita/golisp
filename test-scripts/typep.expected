
t 
t 
t 
t 
nil 
t 
nil 
"====" 
t 
t 
t 
nil 
t 
nil 
"====" 
t 
t 
t 
t 
nil 
"====" 
t 
nil 
"====" 
t 
t 
t 
"====" 
t 
t 
t 
t 
nil 
"====" 
t 
"====" 
t 
